tests/kernel/profiling/profiling_api/src/test_profiling_api.c

```diff
@@ -74,5 +75,38 @@ void test_call_stacks_analyze_workq(void)
 	}
 }
 
+static void thread_analyze(struct k_thread *thread, void *user_data)
+{
+	int *var = (int *)user_data;
+
+	++(*var);
+	TC_PRINT("tid: %p, prio: %d thread_counter: %d\n",
+				thread, thread->base.prio, *var);
+}
+
+static void stack_analyze(const char *stack, size_t size, void *user_data)
+{
+	int *var = (int *)user_data;
+
+	++(*var);
+	TC_PRINT("stack:%p, size:%u stack_counter:%d\n",
+			stack, (unsigned int)size, *var);
+}
+
+void test_k_thread_foreach(void)
+{
+	test_var = 0;
+	k_thread_foreach(thread_analyze, &test_var);
+	zassert_true(test_var, "thread_analyze() not getting called");
```

Comment:
This doesn't prove that it completely works. It passes even if the function just runs once, and not for every thread on the system.
Can you make this stronger, test for a specific value of test_var?